src/main/java/org/opensearch/security/ssl/http/netty/Netty4HttpRequestHeaderVerifier.java

```diff
@@ -0,0 +1,139 @@
+
+import io.netty.channel.ChannelHandler.Sharable;
+import io.netty.channel.ChannelHandlerContext;
+import org.opensearch.http.netty4.Netty4HttpChannel;
+import org.opensearch.http.netty4.Netty4HttpServerTransport;
+import org.opensearch.rest.RestUtils;
+import org.opensearch.security.filter.SecurityRequestChannel;
+import org.opensearch.security.filter.SecurityRequestChannelUnsupported;
+import org.opensearch.security.filter.SecurityRequestFactory;
+import org.opensearch.security.filter.SecurityResponse;
+import org.opensearch.security.filter.SecurityRestFilter;
+import org.opensearch.security.filter.SecurityRestUtils;
+import org.opensearch.security.ssl.transport.SSLConfig;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.security.ssl.OpenSearchSecuritySSLPlugin;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.OpenSearchSecurityException;
+
+import java.util.regex.Matcher;
+
+import static com.amazon.dlic.auth.http.saml.HTTPSamlAuthenticator.API_AUTHTOKEN_SUFFIX;
+import static org.opensearch.security.filter.SecurityRestFilter.HEALTH_SUFFIX;
+import static org.opensearch.security.filter.SecurityRestFilter.PATTERN_PATH_PREFIX;
+import static org.opensearch.security.filter.SecurityRestFilter.WHO_AM_I_SUFFIX;
+import static org.opensearch.security.http.SecurityHttpServerTransport.CONTEXT_TO_RESTORE;
+import static org.opensearch.security.http.SecurityHttpServerTransport.EARLY_RESPONSE;
+import static org.opensearch.security.http.SecurityHttpServerTransport.SHOULD_DECOMPRESS;
+import static org.opensearch.security.http.SecurityHttpServerTransport.IS_AUTHENTICATED;
+
+@Sharable
+public class Netty4HttpRequestHeaderVerifier extends SimpleChannelInboundHandler<DefaultHttpRequest> {
+    private final SecurityRestFilter restFilter;
+    private final ThreadPool threadPool;
+    private final SSLConfig sslConfig;
+    private final boolean injectUserEnabled;
+    private final boolean passthrough;
+
+    public Netty4HttpRequestHeaderVerifier(SecurityRestFilter restFilter, ThreadPool threadPool, Settings settings) {
+        this.restFilter = restFilter;
+        this.threadPool = threadPool;
+
+        this.injectUserEnabled = settings.getAsBoolean(ConfigConstants.SECURITY_UNSUPPORTED_INJECT_USER_ENABLED, false);
+        boolean disabled = settings.getAsBoolean(ConfigConstants.SECURITY_DISABLED, false);
+        if (disabled) {
+            sslConfig = new SSLConfig(false, false);
+        } else {
+            sslConfig = new SSLConfig(settings);
+        }
+        boolean client = !"node".equals(settings.get(OpenSearchSecuritySSLPlugin.CLIENT_TYPE));
+        this.passthrough = client || disabled || sslConfig.isSslOnlyMode();
+    }
+
+    @Override
+    public void channelRead0(ChannelHandlerContext ctx, DefaultHttpRequest msg) throws Exception {
+        // DefaultHttpRequest should always be first and contain headers
```

Comment:
Can you give me some details on this?

Reply:
I think the comment is a little vague, DefaultHttpRequest has headers whereas DefaultFullHttpRequest has content body, so DefaultHttpRequest comes first.  I think we can just remove as the comment doesn't have much bearing on the function contents